### PR TITLE
Adds header options in the manipulate context

### PR DESCRIPTION
**Pull request by dstiliadis**
Add the possibility of custom headers in the manipulate context.

### CODE REVIEW

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "backend",
    "pr-id": "400",
    "commit-sha": "0b75af0f18e3aa456fb5e272551889b5ff35b6e6"
  },
  {
    "project": "",
    "component": "manipulate",
    "pr-id": "81",
    "commit-sha": "ff993f3f6fdb1d94b48b8cbe8842e26872431676"
  }
]
```

File: context.go [hunk 1]
```diff
@@ -123,6 +123,7 @@ type mcontext struct {
 	idempotencyKey       string
 	username             string
 	password             string
+	headers              map[string]string
```
Comment (primalmotion) [inline context.go]: I d)n't like this as header are controlled and this would allow weird shit. Please add a ClientIP property

Comment (primalmotion) [reply context.go]: Plus you don't use them anywhere

Comment (primalmotion) [reply context.go]: Plus yoy should use http.Header

Comment (dstiliadis) [reply context.go]: moved to client IP and its used now

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "manipulate",
    "pr-id": "81",
    "commit-sha": "7aa8d8c4318430e839e562d801f803dad1345df6"
  },
  {
    "project": "",
    "component": "backend",
    "pr-id": "400",
    "commit-sha": "3f4033374e76ef88b29f5fc6e562928246edeac6"
  }
]
```

**Comment by aporeto-dimitri** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "manipulate",
    "pr-id": "81",
    "commit-sha": "7aa8d8c4318430e839e562d801f803dad1345df6"
  },
  {
    "project": "",
    "component": "backend",
    "pr-id": "400",
    "commit-sha": "9f914c966ec5e45ebe401ca7c9f0e7db2b2e6d24"
  }
]
```

**Pull request merged**